modules/renter.go

```diff
@@ -147,6 +148,44 @@ func (s HostDBScans) Len() int           { return len(s) }
 func (s HostDBScans) Less(i, j int) bool { return s[i].Timestamp.Before(s[j].Timestamp) }
 func (s HostDBScans) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
 
+// MerkleRootSet is a set of Merkle roots, and gets encoded more efficiently.
+type MerkleRootSet []crypto.Hash
+
+// MarshalJSON defines a JSON encoding for a MerkleRootSet.
+func (mrs MerkleRootSet) MarshalJSON() ([]byte, error) {
+	// Copy the whole array into a giant byte slice and then encode that.
+	fullBytes := make([]byte, crypto.HashSize*len(mrs))
+	for i := range mrs {
+		copy(fullBytes[i*crypto.HashSize:(i+1)*crypto.HashSize], mrs[i][:])
+	}
+	return json.Marshal(fullBytes)
```

Comment:
lol I also tried doing the base64 manually. Probably should have mentioned that and saved you the trouble.

We could speed it up a lot if we defined MarshalJSON on the whole struct, though that makes compatibility more annoying. But then we get to bypass all of the reflection.